addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -65,6 +68,14 @@ var (
 	useMetrics     = flag.Bool("use-metrics", false, "Whether to use apiserver metrics to detect cluster size instead of the default method of listing objects from the Kubernetes API.")
 	hcAddress      = flag.String("healthcheck-address", ":8080", "The address to expose an HTTP health-check on.")
 	scalingMode    = flag.String("scaling-mode", nanny.NodeProportional, "The mode of scaling to be used. Possible values: 'node-proportional' or 'container-proportional'")
+	// Flags for addon resizer running on the control plane.
+	runOnControlPlane           = flag.Bool("run-on-control-plane", false, "Whether the addon-resizer is running on the control plane.")
+	kubeconfigPath              = flag.String("kubeconfig", "", "absolute path to the kubeconfig file specifying the apiserver instance.")
+	nannyConfigMapName          = flag.String("nanny-config-map-name", "", "The name of the ConfigMap of NannyConfiguration. Specify when getting nanny config updates through API calls.")
+	leaderElectionEnable        = flag.Bool("leader-election-enable", false, "Whether leader election should be used.")
```

Comment:
I tried to follow what vpa does to leverage the existing libs, but when I tried to test, I failed to build the new image - that lib requires go version bump (when I run go mod tidy/vendor) which seems to have some conflicts and complains about undefined resource lock (which is pretty confusing because I didn't change resource lock for election...)

I'm thinking we can keep this as is for now since I was actually also following some internal component examples to implement this part. And we can change to use the lib later. WDYT?